sklearn/utils/tests/test_array_api.py

```diff
@@ -164,6 +167,31 @@ def test_asarray_with_order_ignored():
     assert not X_new_np.flags["F_CONTIGUOUS"]
 
 
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype", yield_namespace_device_dtype_combinations()
+)
+@pytest.mark.parametrize(
+    "sample_weight, normalize, expected",
+    [
+        (None, False, 10.0),
+        (None, True, 2.5),
+        ([0.1, 0.2, 0.3, 0.4], False, 3.0),
+        ([0.1, 0.2, 0.3, 0.4], True, 3.0),
+    ],
+)
+def test_weighted_sum(
+    array_namespace, device, dtype, sample_weight, normalize, expected
+):
+    xp, device, _ = _array_api_for_tests(array_namespace, device, dtype)
+    sample_score = xp.asarray([1, 2, 3, 4])
+
+    with config_context(array_api_dispatch=True):
+        result = _weighted_sum(sample_score, sample_weight, normalize)
+
+    assert isinstance(result, float)
+    assert_allclose(result, expected)
```

Comment:
For consistency:

```suggestion
    assert_allclose(result, expected, atol=np.finfo(dtype).eps * 100)
```

we could even introduce `_array_api_atol(dtype)` to define the expected `atol` for `array_api` compliance tests.